Sources/PitchSpeller/Data Structures/Graph.swift

```diff
@@ -66,32 +71,32 @@ public struct Graph <Value: Hashable>: Hashable {
 
         // MARK: - Instance Properties
 
-        /// The `Graph.Edge` values contained herein.
+        /// The `Graph.Edge` weights contained herein.
         let edges: [Edge]
 
         // MARK: - Initializers
 
-        /// Create a `Graph.Path` with the given array of `Edge` values.
+        /// Create a `Graph.Path` with the given array of `Edge` weights.
         public init(_ edges: [Edge]) {
             self.edges = edges
         }
 
-        /// - Returns: A `Path` with the values of each `Edge` updated by the given `transform`.
-        public func map(_ transform: (Double) -> Double) -> Path {
+        /// - Returns: A `Path` with the weights of each `Edge` updated by the given `transform`.
+        public func map(_ transform: (Weight) -> Weight) -> Path {
             return Path(edges.map { $0.map(transform) })
         }
     }
 
     // MARK: - Instance Properties
 
-    /// - Returns: All of the `Edge` values in the `Graph`.
+    /// - Returns: All of the `Edge` weights in the `Graph`.
     public var edges: [Edge] {
-        return adjacencyList.flatMap { _, values in values }
+        return Array(adjacencyMatrix.values)
     }
 
     /// - Returns: All of the `Node` values in the `Graph`.
     public var nodes: [Node] {
```

Comment:
I would say get rid of this, and rename `nodeSet` to `nodes: Set<Node>`.

Reply:
I agree with you. The thing I'm struggling with is that I want `nodes` to be a Set internally but an Array publicly... Thoughts on just having `nodes` be a public var Set?

Reply:
I would say a `Set` of `nodes` is more correct, and should be the publicly-exposed shape. The `Array` implementation was just what fell out of the adjacency list model.

Reply:
Cool - sounds good 👍